tests/test_app.py

```diff
@@ -184,6 +184,44 @@ def test_nonexist_alias(self):
         body = self.make_request('GET', '/ga4gh/dos/v1/databundles?alias=' + alias)
         self.assertEqual(len(body['data_bundles']), 0)
 
+    def test_full_data_object_update(self):
+        """
+        Demonstrates updating multiple fields of a data object at once.
+        This incidentally also tests object conversion.
+        """
+        # First, select a "random" object that we can test
+        body = self.make_request('GET', '/ga4gh/dos/v1/dataobjects')
+        data_object = body['data_objects'][6]
+        url = '/ga4gh/dos/v1/dataobjects/' + data_object['id']
+
+        # Make a new data object that is different from the data object we retrieved
+        attributes = {
```

Comment:
Are we creating new items in the index here? Can you test that the number of dataobjects is as expected still?

We might consider only allowing overwriting the aliases for now and allowing other modifications to the metadata to come from azul, since we don't have any use cases for it at the time.

Reply:
> Are we creating new items in the index here?

No - though we're making the ES request to a document upsert endpoint (as opposed to a partial document update endpoint as before), `update_data_object` will check first to make sure that the given data object exists before attempting to update it.

> Can you test that the number of dataobjects is as expected still?
Yes - is this something that we just want to check internally, or should we expose it as an endpoint?

> We might consider only allowing overwriting the aliases for now and allowing other modifications to the metadata to come from azul, since we don't have any use cases for it at the time.
That makes sense to me. Overwriting the aliases field is the current behavior (#66) so if so that would mean this issue is all wrapped up!

Reply:
Right, sorry I was looking for this line:

https://github.com/DataBiosphere/dos-azul-lambda/blob/c36835dcd00742d8fde3469f739876cc455e5a8d/app.py#L393